projects/com.oracle.truffle.llvm.parser.factories/src/com/oracle/truffle/llvm/parser/factories/NodeFactoryFacadeImpl.java

```diff
@@ -409,19 +411,42 @@ public LLVMFunctionDescriptor createFunctionDescriptor(String name, LLVMRuntimeT
     }
 
     @Override
-    public Object allocateGlobalVariable(GlobalVariable globalVariable) {
-        if (runtime.isGlobalVariableDefined(globalVariable.getName())) {
+    public LLVMGlobalVariableDescriptor allocateGlobalVariable(GlobalVariable globalVariable) {
+        String linkage = globalVariable.getLinkage();
+        boolean isStatic = "internal".equals(linkage) || "private".equals(linkage);
+        boolean isExtern = "external".equals(linkage);
+
+        String name = globalVariable.getName();
+
+        NativeResolver nativeResolver = new NativeResolver() {
+
+            @Override
+            public LLVMAddress resolve() {
+                return LLVMAddress.fromLong(runtime.getNativeHandle(name));
```

Comment:
The package that `LLVMGlobalVariableDescriptor` is in means that it doesn't have access to things like `LLVMParserRuntime`, so I wasn't sure how to let it resolve the external global variable apart from with this closure.